sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -79,6 +79,25 @@ def test_fit_transform():
                          Y, ridge_alpha=None)
 
 
+def test_fit_transform_variance():
+    alpha = 1
+    rng = np.random.RandomState(0)
+    Y, _, _ = generate_toy_data(3, 10, (8, 8), random_state=rng)  # wide array
+    spca_lars = SparsePCA(n_components=3, method='lars', alpha=alpha,
+                          random_state=0)
+    pca = PCA(n_components=3, random_state=0)
+
+    pca.fit(Y)
+    # no need to fit spca for this
+    spca_lars.fit(Y)
+
+    components = pca.components_
+    explained_variance = pca.explained_variance_
+    explained_variance_sparse = spca_lars.get_explained_variance(components, Y)
+
```

Comment:
why not  ``explained_variance_sparse = spca_lars.explained_variance_`` ?

Reply:
good catch!